### DISCUSSION TITLE
Error on check.msg() on umqtt.simple2 MQTT Package

### DISCUSSION THREAD

**Discussion by DS256** [micropython/micropython#11297] [RP2040 / Pico] [answered]
I am getting the following error on check.msg() with the above package.
`
Traceback (most recent call last):
  File "<stdin>", line 306, in main
  File "/lib/umqtt/simple2.py", line 144, in check_msg
TypeError: function takes 2 positional arguments but 4 were given
`
Line 306 is mqtt.check_msg(). The example code shown in GITHUB has no arguements

Looking at line 144 im simple2.py, it looks like an error thrown by simply2 making a function call.

I haven't found any help on-line

Suggestions? Thank

**Comment by andypiper** [comment]
Where is this code coming from, please?

**Reply by DS256** [reply]
Sorry for the delay. I forgot where I posted this.

To your question, I got the code from here https://github.com/fizista/micropython-umqtt.simple2

**Comment by peterhinch** [comment]
Please see [my reply here](https://github.com/orgs/micropython/discussions/11203#discussioncomment-5541271).

**Reply by DS256** [reply]
I was aware that there was umqtt.simple and umqtt.simple2. I take it the names to not represent a version change but a different product

**Reply by peterhinch** [reply, marked as answer]
They are very different. Unless there is someone here actually using `.simple2` you may have trouble getting support. There are plenty of users of the official library.

**Comment by DS256** [comment]
I was able to get this working by reviewing the same code and realized I had some extra pieces in my calls.
